open-apparel-registry/open-apparel-registry
# Store dedupe model in FileField

## Overview

Stores the TrainedModel.dedupe_model in a FileField. It's stored locally in a .gitignored folder, django_files, during development, and on S3 in production. Files are named `<TrainedModel.id>_dedupe.pickle`.

Connects #2230 

## Demo

<img width="1046" alt="Screen Shot 2022-10-13 at 10 23 31 AM" src="https://user-images.githubusercontent.com/21046714/195623545-997c3c81-7985-4938-9b5c-de12fab6eb7c.png">

## Notes

I moved the FacilityList.file to the `django_files` folder at the same time. 

## Testing Instructions

* Run `./scripts/resetdb`. This should work without error. There should be a file in `django_files/models/` called `1_dedupe.pickle`.
* Run `./scripts/manage train_model`. This should work without error. There should be a file in `django_files/models/` called `2_dedupe.pickle`.
* Set the environment to use S3:
     * In `docker-compose.yml` set `AWS_STORAGE_BUCKET_NAME=opensupplyhub-testing-files-eu-west-1`
     * In `settings.py` change line 462 from `if not DEBUG:` to `if True`
* Repeat the above steps. Everything should work as before. If you go to `opensupplyhub-testing-files-eu-west-1` in S3, you will see the model files uploaded. 
* Upload a FacilityList and then use the download button to download the raw file to confirm that's working as before. 

## Checklist
